index.bs

```diff
@@ -535,167 +738,319 @@ To determine whether an |attribute| <dfn>contains a javascript: URL</dfn>:
-    - [=list/size=] of |configuration|["{{SanitizerConfig/elements}}"] equals
-      [=list/size=] of |sanitizer|'s [=Sanitizer/configuration=]["{{SanitizerConfig/elements}}"].
-    - [=list/size=] of |configuration|["{{SanitizerConfig/removeElements}}"] equals
-      [=list/size=] of |sanitizer|'s [=Sanitizer/configuration=]["{{SanitizerConfig/removeElements}}"].
-    - [=list/size=] of |configuration|["{{SanitizerConfig/replaceWithChildrenElements}}"] equals
-      [=list/size=] of |sanitizer|'s [=Sanitizer/configuration=]["{{SanitizerConfig/replaceWithChildrenElements}}"].
-    - [=list/size=] of |configuration|["{{SanitizerConfig/attributes}}"] equals
-      [=list/size=] of |sanitizer|'s [=Sanitizer/configuration=]["{{SanitizerConfig/attributes}}"].
-    - [=list/size=] of |configuration|["{{SanitizerConfig/removeAttributes}}"] equals
-      [=list/size=] of |sanitizer|'s [=Sanitizer/configuration=]["{{SanitizerConfig/removeAttributes}}"].
-    - Either |configuration|["{{SanitizerConfig/elements}}"] or
-      |configuration|["{{SanitizerConfig/removeElements}}"] [=map/exist=],
-      or neither, but not both.
-    - Either |configuration|["{{SanitizerConfig/attributes}}"] or
-      |configuration|["{{SanitizerConfig/removeAttributes}}"] [=map/exist=],
-      or neither, but not both.
-
-Note: Previous versions of this spec had elaborate definitions of how to
-  canonicalize a config. This has now effectively been moved into the method
-  definitions.
-
-Note: This operation is defined in terms of the manipulation methods on the
-    {{Sanitizer}}. Those methods remove matching entries from other lists.
-    The size equality steps in the last step would then catch this.
-    For example:
-    `{ allow: ["div", "div"] }` would create a Sanitizer with one element in
-    the allow list. The final test would then return false, which would cause
-    the caller to throw an exception.
-
-Issue: This is still missing error checks for the per-element attribute lists
-    and syntax errors.
+To <dfn for="Sanitizer">set a configuration</dfn>, given a [=dictionary=] |configuration|,
+a [=boolean=] |allowCommentsAndDataAttributes|, and a {{Sanitizer}} |sanitizer|:
+
+1. [=Canonicalize the configuration|Canonicalize=] |configuration| with
+    |allowCommentsAndDataAttributes|.
+1. If |configuration| is not [=SanitizerConfig/valid=], then return false.
+1. Set |sanitizer|'s [=Sanitizer/configuration=] to |configuration|.
+1. Return true.
+
+</div>
+
+## Canonicalize the Configuration ## {#configuration-canonicalize}
+
+The {{Sanitizer}} stores the [=Sanitizer/configuration=] in a canonical form, as this makes
+a number of processing steps easier.
+
+<div class=example>An {{SanitizerConfig/elements}} list `{elements: ["div"]}` gets stored as
+`{elements: [{name: "div", namespace: "http://www.w3.org/1999/xhtml"}]`).
+</div>
+
+<div algorithm>
+To <dfn for=Sanitizer>canonicalize the configuration</dfn> {{SanitizerConfig}} |configuration|
```

Comment:
Depending on your definition of *canonical*, I would say the result is not necessarily in *canonical* form, e.g. has duplicates across different `elements` and `removeElements` etc.